Bump commons-ui v0.150 + MUI X migration v7 to v8

flomillot opened the pull request:
## PR Summary
Updated MUI X to version 8.
Refactoring of CustomTreeItem.
Cleaning styles.

flomillot committed:
Update MUI X to v8

Signed-off-by: Florent MILLOT <75525996+flomillot@users.noreply.github.com>

flomillot committed:
Merge branch 'main' into update-mui-x-v8

Mathieu-Deharbe reviewed (changes requested):
The way you change the hover/selection has side effects : 

The arrows that show if there are subdirectories appears only when clicked : 
![fèche seulement si clique](https://github.com/user-attachments/assets/4489a637-2e67-47d8-aca7-694dfe5a5d56)

The hover and selection effect on the line is now different. I like it like this but did you check with Stéphane ? : 

before : 
![hover avant](https://github.com/user-attachments/assets/1134c9d9-83d3-42c3-9ac4-472643e19932)
now : 
![hover now](https://github.com/user-attachments/assets/0cb39dc0-e4ab-461c-92b9-4410cb9ff5cd)

Mathieu-Deharbe committed:
Merge branch 'main' into update-mui-x-v8

flomillot committed:
Refactor tree view components to improve utility functions

Replaced redundant icon components with direct styling via `slotProps`. Introduced `doesNodeHasChildren` utility function for cleaner child checks across tree components. Added placeholders to ensure proper expand icons rendering in parent nodes without immediate children.

Signed-off-by: Florent MILLOT <75525996+flomillot@users.noreply.github.com>

flomillot commented:
> The way you change the hover/selection has side effects :
> 
> The arrows that show if there are subdirectories appears only when clicked : ![fèche seulement si clique](https://private-user-images.githubusercontent.com/148252167/527127817-4489a637-2e67-47d8-aca7-694dfe5a5d56.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjYwNTIwNTYsIm5iZiI6MTc2NjA1MTc1NiwicGF0aCI6Ii8xNDgyNTIxNjcvNTI3MTI3ODE3LTQ0ODlhNjM3LTJlNjctNDdkOC1hY2E3LTY5NGRmZTVhNWQ1Ni5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxOFQwOTU1NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMjc0MDVjMTY1NGJiMjliMmE1Yzk0MDM3ZTViZGRkMjlmY2FhMmQyNGQ3NWFmY2M5NGMzOGIzNDZhMTY0ODlmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.34RS3nr1vSlK8hFU7lTj4WsJLtZyKiEyr_Sa3nb0ji4) [ ![fèche seulement si clique](https://private-user-images.githubusercontent.com/148252167/527127817-4489a637-2e67-47d8-aca7-694dfe5a5d56.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjYwNTIwNTYsIm5iZiI6MTc2NjA1MTc1NiwicGF0aCI6Ii8xNDgyNTIxNjcvNTI3MTI3ODE3LTQ0ODlhNjM3LTJlNjctNDdkOC1hY2E3LTY5NGRmZTVhNWQ1Ni5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxOFQwOTU1NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMjc0MDVjMTY1NGJiMjliMmE1Yzk0MDM3ZTViZGRkMjlmY2FhMmQyNGQ3NWFmY2M5NGMzOGIzNDZhMTY0ODlmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.34RS3nr1vSlK8hFU7lTj4WsJLtZyKiEyr_Sa3nb0ji4) ](https://private-user-images.githubusercontent.com/148252167/527127817-4489a637-2e67-47d8-aca7-694dfe5a5d56.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjYwNTIwNTYsIm5iZiI6MTc2NjA1MTc1NiwicGF0aCI6Ii8xNDgyNTIxNjcvNTI3MTI3ODE3LTQ0ODlhNjM3LTJlNjctNDdkOC1hY2E3LTY5NGRmZTVhNWQ1Ni5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxOFQwOTU1NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMjc0MDVjMTY1NGJiMjliMmE1Yzk0MDM3ZTViZGRkMjlmY2FhMmQyNGQ3NWFmY2M5NGMzOGIzNDZhMTY0ODlmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.34RS3nr1vSlK8hFU7lTj4WsJLtZyKiEyr_Sa3nb0ji4) [ ](https://private-user-images.githubusercontent.com/148252167/527127817-4489a637-2e67-47d8-aca7-694dfe5a5d56.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjYwNTIwNTYsIm5iZiI6MTc2NjA1MTc1NiwicGF0aCI6Ii8xNDgyNTIxNjcvNTI3MTI3ODE3LTQ0ODlhNjM3LTJlNjctNDdkOC1hY2E3LTY5NGRmZTVhNWQ1Ni5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxOFQwOTU1NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMjc0MDVjMTY1NGJiMjliMmE1Yzk0MDM3ZTViZGRkMjlmY2FhMmQyNGQ3NWFmY2M5NGMzOGIzNDZhMTY0ODlmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.34RS3nr1vSlK8hFU7lTj4WsJLtZyKiEyr_Sa3nb0ji4)
> 
> The hover and selection effect on the line is now different. I like it like this but did you check with Stéphane ? :
> 
> before : ![hover avant](https://private-user-images.githubusercontent.com/148252167/527128868-1134c9d9-83d3-42c3-9ac4-472643e19932.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjYwNTIwNTYsIm5iZiI6MTc2NjA1MTc1NiwicGF0aCI6Ii8xNDgyNTIxNjcvNTI3MTI4ODY4LTExMzRjOWQ5LTgzZDMtNDJjMy05YWM0LTQ3MjY0M2UxOTkzMi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxOFQwOTU1NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MzYxY2QyZDgzN2M4NTNlYjI1Mjk5ZDU3NDNjYWJiMmFiMjlhNmQyOTAwMDg4OThmNTE1MjQxOTY3ZjI4ZWY4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.LcZMsbIpTE3LWuSt_C7J_jmyCcnA2V0hTCS5NZs8YmI) [ ![hover avant](https://private-user-images.githubusercontent.com/148252167/527128868-1134c9d9-83d3-42c3-9ac4-472643e19932.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjYwNTIwNTYsIm5iZiI6MTc2NjA1MTc1NiwicGF0aCI6Ii8xNDgyNTIxNjcvNTI3MTI4ODY4LTExMzRjOWQ5LTgzZDMtNDJjMy05YWM0LTQ3MjY0M2UxOTkzMi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxOFQwOTU1NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MzYxY2QyZDgzN2M4NTNlYjI1Mjk5ZDU3NDNjYWJiMmFiMjlhNmQyOTAwMDg4OThmNTE1MjQxOTY3ZjI4ZWY4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.LcZMsbIpTE3LWuSt_C7J_jmyCcnA2V0hTCS5NZs8YmI) ](https://private-user-images.githubusercontent.com/148252167/527128868-1134c9d9-83d3-42c3-9ac4-472643e19932.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjYwNTIwNTYsIm5iZiI6MTc2NjA1MTc1NiwicGF0aCI6Ii8xNDgyNTIxNjcvNTI3MTI4ODY4LTExMzRjOWQ5LTgzZDMtNDJjMy05YWM0LTQ3MjY0M2UxOTkzMi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxOFQwOTU1NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MzYxY2QyZDgzN2M4NTNlYjI1Mjk5ZDU3NDNjYWJiMmFiMjlhNmQyOTAwMDg4OThmNTE1MjQxOTY3ZjI4ZWY4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.LcZMsbIpTE3LWuSt_C7J_jmyCcnA2V0hTCS5NZs8YmI) [ ](https://private-user-images.githubusercontent.com/148252167/527128868-1134c9d9-83d3-42c3-9ac4-472643e19932.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjYwNTIwNTYsIm5iZiI6MTc2NjA1MTc1NiwicGF0aCI6Ii8xNDgyNTIxNjcvNTI3MTI4ODY4LTExMzRjOWQ5LTgzZDMtNDJjMy05YWM0LTQ3MjY0M2UxOTkzMi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxOFQwOTU1NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MzYxY2QyZDgzN2M4NTNlYjI1Mjk5ZDU3NDNjYWJiMmFiMjlhNmQyOTAwMDg4OThmNTE1MjQxOTY3ZjI4ZWY4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.LcZMsbIpTE3LWuSt_C7J_jmyCcnA2V0hTCS5NZs8YmI) now : ![hover now](https://private-user-images.githubusercontent.com/148252167/527128920-0cb39dc0-e4ab-461c-92b9-4410cb9ff5cd.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjYwNTIwNTYsIm5iZiI6MTc2NjA1MTc1NiwicGF0aCI6Ii8xNDgyNTIxNjcvNTI3MTI4OTIwLTBjYjM5ZGMwLWU0YWItNDYxYy05MmI5LTQ0MTBjYjlmZjVjZC5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxOFQwOTU1NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MDk1NmFjZTRlNDBiMzQ3OGY5MDE0NTdiN2UyZDA4Y2Y5NTQ0Mjg3ZDJmMzhjYzBmZTFjMTMwY2I3MTIyNTU0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.IJqW3Fw9IS6SmddKOyRhfbcz91TRKd39gl4E2BWe_18) [ ![hover now](https://private-user-images.githubusercontent.com/148252167/527128920-0cb39dc0-e4ab-461c-92b9-4410cb9ff5cd.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjYwNTIwNTYsIm5iZiI6MTc2NjA1MTc1NiwicGF0aCI6Ii8xNDgyNTIxNjcvNTI3MTI4OTIwLTBjYjM5ZGMwLWU0YWItNDYxYy05MmI5LTQ0MTBjYjlmZjVjZC5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxOFQwOTU1NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MDk1NmFjZTRlNDBiMzQ3OGY5MDE0NTdiN2UyZDA4Y2Y5NTQ0Mjg3ZDJmMzhjYzBmZTFjMTMwY2I3MTIyNTU0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.IJqW3Fw9IS6SmddKOyRhfbcz91TRKd39gl4E2BWe_18) ](https://private-user-images.githubusercontent.com/148252167/527128920-0cb39dc0-e4ab-461c-92b9-4410cb9ff5cd.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjYwNTIwNTYsIm5iZiI6MTc2NjA1MTc1NiwicGF0aCI6Ii8xNDgyNTIxNjcvNTI3MTI4OTIwLTBjYjM5ZGMwLWU0YWItNDYxYy05MmI5LTQ0MTBjYjlmZjVjZC5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxOFQwOTU1NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MDk1NmFjZTRlNDBiMzQ3OGY5MDE0NTdiN2UyZDA4Y2Y5NTQ0Mjg3ZDJmMzhjYzBmZTFjMTMwY2I3MTIyNTU0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.IJqW3Fw9IS6SmddKOyRhfbcz91TRKd39gl4E2BWe_18) [ ](https://private-user-images.githubusercontent.com/148252167/527128920-0cb39dc0-e4ab-461c-92b9-4410cb9ff5cd.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjYwNTIwNTYsIm5iZiI6MTc2NjA1MTc1NiwicGF0aCI6Ii8xNDgyNTIxNjcvNTI3MTI4OTIwLTBjYjM5ZGMwLWU0YWItNDYxYy05MmI5LTQ0MTBjYjlmZjVjZC5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxOFQwOTU1NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MDk1NmFjZTRlNDBiMzQ3OGY5MDE0NTdiN2UyZDA4Y2Y5NTQ0Mjg3ZDJmMzhjYzBmZTFjMTMwY2I3MTIyNTU0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.IJqW3Fw9IS6SmddKOyRhfbcz91TRKd39gl4E2BWe_18)

Yes, I didn't see it it's fixed now, thank you.

flomillot committed:
Fix tree item focus styling by updating CSS selector for MUI TreeItem

Signed-off-by: Florent MILLOT <75525996+flomillot@users.noreply.github.com>

flomillot committed:
Merge branch 'main' into update-mui-x-v8

flomillot committed:
Update commons ui

Signed-off-by: Florent MILLOT <florent.millot_externe@rte-france.com>
